src/main/java/parser/ParserImpl.java

```diff
@@ -0,0 +1,225 @@
+	 */
+	public ParserImpl(ICommandLogicUnit commandLogicUnit, IUi ui) {
+		this.commandLogicUnit = commandLogicUnit;
+		this.ui = ui;
+	}
+	
+	@Override
+	public void processInput(String input) {
+		// take the next command within a line, inputted using enter
+		List<String> parsedCommands = Arrays.asList(input.split(" "));
+		
+		processInputHelper(parsedCommands);
+	}
+	
+	/**
+	 * Processes the input as list of strings into commands.
+	 * Helper method for processInput method.
+	 *
+	 * @param parsedCommands List of string from user inputs.
+	 */
+	private void processInputHelper(List<String> parsedCommands) {
+		switch (parsedCommands.get(0)) {
+		case "bye":
+			commandLogicUnit.processCommand(Command.BYE, Map.of());
+			break;
+		case "list":
+			parseList(parsedCommands);
+			break;
+		case "done":
+			parseDone(parsedCommands);
+			break;
+		case "deadline":
+			parseDeadline(parsedCommands);
+			break;
+		case "todo":
+			parseTodo(parsedCommands);
+			break;
+		case "event":
+			parseEvent(parsedCommands);
+			break;
+		case "delete":
+			parseDelete(parsedCommands);
+			break;
+		case "find":
+			parseFind(parsedCommands);
+			break;
+		case "man":
+			ui.printSentence("() -> optional parameter or format of date time\n" +
+					"# -> the parameter of the commands");
+			ui.printIndexedList(availableCommands);
+			break;
+		default:
+			logger.warning("Unknown command : " + parsedCommands.get(0) + "being parsed");
+			processException(new IllegalCallerException("I'm sorry, but I don't know what that means :-("));
+			break;
+		}
+	}
+	
+	private void parseList(List<String> parsedCommands) {
+		String date = "";
+		
+		if (parsedCommands.size() >= 2) {
+			date = parsedCommands.get(1);
+		}
+		
+		commandLogicUnit.processCommand(Command.LIST, Map.of(
+				"date", date
+		));
+	}
+	
+	private void parseDone(List<String> parsedCommands) {
+		try {
+			commandLogicUnit.processCommand(Command.DONE, Map.of(
+					"index", parsedCommands.get(1)
+			));
+		} catch (Exception e) {
+			processException(e);
+		}
+	}
+	
+	private void parseDeadline(List<String> parsedCommands) {
+		try {
+			int byIndex = parsedCommands.indexOf("/by");
+			if (byIndex == -1) {
+				logger.warning("/by not supplied in deadline command");
+				throw new IllegalArgumentException("/by command not found");
+			}
+			
+			String desc = String.join(" ", parsedCommands.subList(1, byIndex));
+			if (desc.isBlank()) {
+				logger.warning("Empty deadline desc supplied");
+				throw new IllegalArgumentException("deadline desc cannot be empty");
+			}
+			
+			String timing = String.join(" ", parsedCommands.subList(byIndex + 1, parsedCommands.size()));
+			if (timing.isBlank()) {
+				logger.warning("Empty deadline timing supplied");
+				throw new IllegalArgumentException("deadline timing cannot be empty");
+			}
+			
+			commandLogicUnit.processCommand(Command.DEADLINE, Map.of(
+					"description", desc,
+					"timing", timing
+			));
+		} catch (Exception e) {
```

Comment:
Perhaps you can specify which Exception you are handling, and not just generalise it to all exceptions.

Reply:
Yeah sure! Resolved